array_api_tests/test_linalg.py

```diff
@@ -645,11 +644,52 @@ def test_vecdot(dtypes, shape, data):
     # TODO: assert shape and elements
 
 
+# Insanely large orders might not work. There isn't a limit specified in the
+# spec, so we just limit to reasonable values here.
+max_ord = 100
+
 @pytest.mark.xp_extension('linalg')
 @given(
-    x=xps.arrays(dtype=xps.floating_dtypes(), shape=shapes()),
-    kw=kwargs(axis=todo, keepdims=todo, ord=todo)
+    x=xps.arrays(dtype=xps.floating_dtypes(), shape=shapes(min_side=1)),
+    data=data(),
 )
-def test_vector_norm(x, kw):
-    # res = linalg.vector_norm(x, **kw)
-    pass
+def test_vector_norm(x, data):
+    kw = data.draw(
+        # We use data because axes is parameterized on x.ndim
+        kwargs(axis=axes(x.ndim),
+               keepdims=booleans(),
+               ord=one_of(
+                   just(1),
+                   just(2),
+                   just(float('inf')),
+                   integers(1, max_ord),
+                   floats(1, max_ord, allow_nan=False,
+                          allow_infinity=False),
+                   just(0),
+                   just(-1),
+                   just(-2),
+                   just(float('-inf')),
+                   integers(-max_ord, -1),
+                   floats(-max_ord, -1, allow_nan=False,
+                          allow_infinity=False),
+               )), label="kw")
```

Comment:
How about using:
```python
ord=one_of(
    integers(-max_ord, max_ord),
    floats(-max_ord, max_ord),
    sampled_from([float("inf"), float("-inf")]),
)
```
Note setting both bounds in `st.floats()` will disable generation of NaNs and infinities. Were you concerned about minimisation behaviour?